source/server/http_test_server_filter.h

```diff
@@ -2,20 +2,68 @@
 
 #include <string>
 
+#include "envoy/common/time.h"
 #include "envoy/server/filter_config.h"
 
+#include "external/envoy/source/common/common/lock_guard.h"
+#include "external/envoy/source/common/common/thread.h"
+
 #include "api/server/response_options.pb.h"
 
 namespace Nighthawk {
 namespace Server {
 
-// Basically this is left in as a placeholder for further configuration.
+/**
+ * Filter configuration container class for the test server extension.
+ * Instances of this class will be shared accross instances of HttpTestServerDecoderFilter.
+ */
 class HttpTestServerDecoderFilterConfig {
 public:
+  /**
+   * Constructs a new HttpTestServerDecoderFilterConfig instance.
+   *
+   * @param proto_config The proto configuration of the filter.
+   */
   HttpTestServerDecoderFilterConfig(nighthawk::server::ResponseOptions proto_config);
+
+  /**
+   * @return const nighthawk::server::ResponseOptions& read-only reference to the proto config
+   * object.
+   */
   const nighthawk::server::ResponseOptions& server_config() { return server_config_; }
 
+  /**
+   * Gets the number of elapsed nanoseconds since the last call (server wide).
```

Comment:
How useful is this statistic if it is server wide as compared to having it per each session? I likely don't fully understand the use case for this statistic, so this may as well be working as intended.

Reply:
The way I see it is that the current approach is going to be particularly useful in A/B tests, for example:

with and without an intermediary
with nighthawk client features that might perform relatively heavy lifting on cpu cores shared with the workers (e.g. stats flushing).
across different clients

So the use case is sanity checking and exposing a low effort means to dig up a potential root cause of latency measurement divergence.

I think per-worker or per-session tracking could also serve as a purpose to further dig into things as that would allow one to get a sense of balancedness across threads and sessions.  Also see https://github.com/envoyproxy/nighthawk/pull/477/files#r476803233 for an earlier discussion related to this.